src/mlpack/methods/ann/layer/linear3d.hpp

```diff
@@ -0,0 +1,180 @@
+/**
+ * @file methods/ann/layer/linear3d.hpp
+ * @author Mrityunjay Tripathi
+ *
+ * Definition of the Linear layer class which accepts 3D input.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_ANN_LAYER_LINEAR3D_HPP
+#define MLPACK_METHODS_ANN_LAYER_LINEAR3D_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/methods/ann/layer/layer_types.hpp>
+#include <mlpack/methods/ann/regularizer/no_regularizer.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * Implementation of the Linear3D layer class. The Linear class represents a
+ * single layer of a neural network.
+ *
```

Comment:
Could you describe here the shape of the input and the output? It could be very helpful.

Reply:
Sure.